compile/service_test.go

```diff
@@ -142,6 +177,16 @@ func TestCompileService(t *testing.T) {
 			keyValueSpec,
 		},
 		{
+			"service annotations",
+			`
+				service AnnotatedService {
+					void setValue(1: string key, 2: binary value) (test = "ok")
+				} (test = "test")
```

Comment:
suggest having tests where the key and value are different so we're not accidentally swapping them anywhere.